### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by mvanputgrccontrol** [LSXPrime/Aegis#1]
Thank you for creating, building and publishing this library. It looks really nice!
Will test the functionality later on. I'm currently using my custom library for licensing our product. 

The feedback I have (also based on requirements I face(d)) based on code review.
- Support for multi tenancy. Load license information based on the tenant provided. For example based on the tenant URL, tenant header of user claims.
- Integrate feature management with `Microsoft.FeatureManagement.AspNetCore` (or other feature management libraries). Feature flags can be loaded in to the `IFeatureManager`. Same applies for rate limit support.
- For the Aegis.Server, I would suggest creating a library for this, so you can add it to the any existing/new ASP.NET Core Project. You can then register it using `builder.AddAegisServer(..)` for example. This allows you to update the server logic and integrate with oidc authorization etc. Another solution would be to create a `dotnet new template` for this. 

Keep up the good work. I'm happy to contribute and discuss to this project!

**Comment by LSXPrime** [comment]
Thank you so much for the positive feedback and the code review! I really appreciate you taking the time to look through the project.  I'm thrilled you see the potential in Aegis!

Your suggestions are excellent and align with my vision of making Aegis even more powerful and flexible. 

I've already started brainstorming how to incorporate:

- **Multi-tenancy support:**  I am not relative with multi-tenancy services yet, using .NET for 8 months now so multi-tenancy was ahead my roadmap, so I would like to ask for explanation on how it might benefit the library and an example or the first steps to apply it and I would be happy to integrate it.
- **Integration with Feature Management Libraries:**  I created Aegis to act as alternative to Portable.Licensing to enable basic functionality, so I find using the default feature management enough, but like most of my projects it gets stars but no one really use it, but if someone really gonna contribute or use this project, I would extend the default features management functionality and Integrate `Microsoft.FeatureManagement.AspNetCore` . Mapping Aegis features to a feature manager would be a seamless way to enable or disable functionality based on licenses, but I am hesitant to use it as main features manager or keep the default implementation and integrate it using separated package aka `Aegis.Microsoft.FeatureManagement`.
- **Aegis.Server as a Library/Template:** Absolutely agree!  Having Aegis.Server as a standalone library that can be easily added to existing ASP.NET Core projects would make it much more versatile.  I'm leaning towards the `builder.AddAegisServer(...)` approach as it provides nice extensibility.  The `dotnet new` template is also a great idea for quickly scaffolding a new Aegis.Server project but I have no experience for creating .net templates and it's less flexible to build a whole project and some new unknown codebase rather than using it as just addition.

I'm very open to contributions and discussions! Would you be interested in collaborating on any of these features?  It would be fantastic to have your input as we work on enhancing Aegis.

Thanks again for your support!

**Comment by mvput** [comment]
I'm happy to collaborate for those features. 

Regarding the integration with other libraries, I totally agree with your comments. My idea also is to create extension packages for example `Microsoft.FeatureManagement`. However this is a nice to have for long term. 

For the other ones, the Aegis Server library is the best step to start. 

Do you have any contribution guidelines? 
We can also connect on slack or something.

**Comment by LSXPrime** [comment]
I don't have a specific contribution guidelines yet.

well, it's always good to have plans for long term, and I agree that the server library is the best to start with.
for Aegis Server library I planned to strip out all asp.net dependencies and just embed the `LicenseService` and `HeartbeatMonitorService` and make the developers inherit their `DbContext` on top of `Aegis DbContext` to have all necessary tables, and let them create their own controllers and auth services and provide the endpoints to the core library, any suggestions on that are welcome.

And I would like to connect, although I usually use FB or reddit only, they aren't teams targeted, I compared slack and discord and I found that discord is more feature rich than slack, if you were open to connect on discord I would be happy, my username is `_lsxprime` and we got a server at [Discord](https://discord.gg/7XGeJ7kZWA)
